src/shared/cryptsetup-fido2.c

```diff
@@ -48,27 +49,21 @@ int acquire_fido2_key(
         assert(cid);
         assert(key_file || key_data);
 
-        if (key_data) {
-                salt = key_data;
-                salt_size = key_data_size;
-        } else {
-                _cleanup_free_ char *bindname = NULL;
-
-                /* If we read the salt via AF_UNIX, make this client recognizable */
-                if (asprintf(&bindname, "@%" PRIx64"/cryptsetup-fido2/%s", random_u64(), volume_name) < 0)
-                        return log_oom();
-
-                r = read_full_file_full(
-                                AT_FDCWD, key_file,
-                                key_file_offset == 0 ? UINT64_MAX : key_file_offset,
-                                key_file_size == 0 ? SIZE_MAX : key_file_size,
-                                READ_FULL_FILE_CONNECT_SOCKET,
-                                bindname,
-                                (char**) &loaded_salt, &salt_size);
+        if (key_data)
+                salt = &IOVEC_MAKE(key_data, key_data_size);
```

Comment:
this does not work, since IOVEC_MAKE returns a compound initialized structure, i.e. something from the stack that gets freed when the scope is left, i.e. already when this if expression ends. Thus `salt` will point to invalid memory before we can use it.

Reply:
Thanks, very useful. This is the first C project I'm contributing to, and coming from high-level languages I'm still trying to grasp the memory management. Should be fixed now.